packages/react/src/auth/useSession.ts

```diff
@@ -0,0 +1,37 @@
+import { useApi } from "../../src/GadgetProvider";
+import { useGet } from "../../src/useGet";
+
+export interface GadgetSession {
+  id: string;
+  userId: string | null;
+  user: GadgetUser | null;
+  [key: string]: any;
+}
+
+export interface GadgetUser {
```

Comment:
Dang, I didn't realize this before, but putting the hooks in this package means they can't reference the types of the actual upstream Gadget models, so this type is not super useful. I think that if the hook below was actually passed an api client where the type of that client could be inferred, it could compute the return type properly, but it doesn't get the api client's type because it comes from `useApi`. Dang.

Reply:
I can avoid exporting `GadgetUser`, but the implementation currently has dependencies on the field `userId`. We could look at moving it into core if it lets us build better types, but do you think this should ship as is?